### PR TITLE
Yamcs search

### PR DESCRIPTION
**Pull request by davetsay**
Reviewer Note: `search` function in `object-provider.js` in is the only new functionality. All other changes are linting changes.

## Overview
Addresses #31 

## Testing Instructions 
see #31 

## Author Checklist
| | |
| --- | :---: |
| Changes address original issue? | Y |
| Unit tests included and/or updated with changes? | N/A |
| Command line build passes? | Y |
| Changes have been smoke-tested? | Y |
| Testing instructions included? | Y |

### CODE REVIEW

**Comment by davetsay** [comment]
@akhenry , ready for review alongside with https://github.com/nasa/openmct/pull/3610

**Review by akhenry — APPROVED** [review]
@davetsay  This looks good to go once the API changes have been merged in Open, thanks!

**Pull request merged**
